### ISSUE TITLE
Regression: Websocket fails to reconnect if it's ever disconnected

### ISSUE THREAD

**Issue by elimisteve** [cryptag/leapchat#319]
@jimmcgaw This means that if we redeploy, the app breaks until the user refreshes, not just for 10 seconds!

```
main_3e302b8450a9a3d6379f.bundle.js:2 Websocket got close event CloseEvent {isTrusted: true, wasClean: false, code: 1006, reason: '', type: 'close', …}
onWsClose @ main_3e302b8450a9a3d6379f.bundle.js:2
main_3e302b8450a9a3d6379f.bundle.js:2 Error: The websocket connection was closed!
    at WebSocket.onWsClose (main_3e302b8450a9a3d6379f.bundle.js:2:1948905)
```

Originally there was logic in there that would set everything to `null` then basically create a new WebSocket connection `onClose`. Now an exception is thrown and isn't recovered from.

**Comment by elimisteve** [comment]
I don't think that logic is gone, it's just not being triggered or something

**Comment by elimisteve** [comment]
I think I see how to fix this... I think when Epics were added, the reconnect logic was removed. Re-adding...

**Comment by elimisteve** [comment]
Hmm, I've still got a browser tab open from like 2 weeks ago -- epics are there, more recent code is not -- and it reconnects to the backend just fine. So something pretty recent is preventing the reconnect from working.

**Comment by jimmcgaw** [comment]
Do you see the `'Lost connection to chat server. Trying to reconnect...'` alert message?

**Comment by elimisteve** [comment]
If I connect to localhost:8080 , then kill the local backend, the frontend doesn't reconnect and stays in this state:

![leapchat-websocket-reconnect-doesnt-Screenshot_2023-02-28_21-05-48](https://user-images.githubusercontent.com/139603/222050054-b7df39dc-ecce-48b4-8762-ac81140da174.png)

**Comment by elimisteve** [comment]
> Do you see the `'Lost connection to chat server. Trying to reconnect...'` alert message?

Yes I do, and the UI keeps showing that yellow banner; it never gets replaced with the green one saying it successfully connected to the server.

**Comment by elimisteve** [comment]
In the successfully-reconnecting old tab, I still get

> Error: Error occurred on websocket connection!

and

> Websocket got close event close { target: WebSocket, isTrusted: true, wasClean: false, code: 1006, reason: "", srcElement: WebSocket, currentTarget: WebSocket, eventPhase: 2, bubbles: false, cancelable: false, … }

exceptions, but those don't prevent a successful reconnection afterward.

**Comment by elimisteve** [comment]
Fixed in #321
